statistics-reports.md

```diff
@@ -0,0 +1,206 @@
+# Displaying usage statistics
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Statistics for a DSpaceObject
+**GET /api/statistics/usagereports**
+
+As we don't have a use case to iterate over all the existing usage reports, the main endpoint is not implemented and a 405 error code is returned according to our [general error response codes](README.md#Error-codes).
+
+## Single statistic
+**GET /api/statistics/usagereports/<:id>**
+
+This endpoint provides a specific statistic
+
+An example JSON response document to `/api/statistics/usagereports/1911e8a4-6939-490c-b58b-a5d70f8d91fb_TopCountries`:
+```json
+{
+    "id": "1911e8a4-6939-490c-b58b-a5d70f8d91fb_TopCountries",
+    "type": "usagereport",
+    "report-type": "TopCountries",
+    "points": [
+        {
+            "label": "United States",
+            "type": "country",
+            "id": "US",
+            "values": {
+                "views": 2
+            }
+        },
+        {
+            "label": "China",
+            "type": "country",
+            "id": "CN",
+            "values": {
+                "views": 1
+            }
+        }
+    ]
+}
+```
+
+The ID is built from a concatenation of the UUID and the type of report for the given UUID
+
+The type of point can be:
+* country
+* city
+* bitstream
+* item
+* collection
+* community
+* date
+
+The report-type specifies what data is returned, relevant e.g. to add a label to report. The current statistics support:
+* TotalVisits
+* TotalVisitsPerMonth
+* TotalDownloads
+* TopCountries
+* TopCities
+
+Possible response status
+
+- 200 OK - The specific statistics data was found, and the data has been properly returned.
+* 401 Unauthorized - if you are not authenticated and the statistics are not available to the Anonymous user
+* 403 Forbidden - if you are not logged in with sufficient permissions and the statistics are not available to the Anonymous user
+- 404 Not Found - The specified ID was not found
+
+## Search Statistics for a DSpaceObject
+**GET /api/statistics/usagereports/search/object**
+
+This endpoint provides a paginated list of statistics for a DSpaceObject. 
+
+The DSpaceObject is given through the following parameters:
+- `scope` The UUID of the DSpaceObject
+- `scopeType` The string type of the DSpaceObject 
+
+The usual parameters for paginated lists are supported as well:
+- `page` The page number 
+- `size` The number of reports in a page
+
+An example JSON response document to `/api/statistics/usagereports/search/object?page=0&size=2&scopeType=site&scope=6d65c6a2-3fe7-44dd-bacb-79271257c35d`:
```

Comment:
Is there a reason you chose `scope` + `scopeType` to represent the object, instead of just using `uri` here?  As I noted previously, this endpoint is very similar to this authorizations endpoint (which uses `uri`): https://github.com/DSpace/Rest7Contract/blob/master/authorizations.md#object

So, I was expecting this to be `/api/statistics/usagereports/search/object?uri=[full-uri-of-object]&page=0&size=2`

Reply:
The reason for the 2 separate parts was simply because that's how the data is stored as well.
But there's no need to map it so closely to the storage, and the URI syntax can definitely be used for consistency reasons. I'll adjust it accordingly